### ISSUE TITLE
LRN operator doesn't comply with ONNX specification

### ISSUE THREAD

**Issue by matteosal** [apache/incubator-mxnet#11281]
ONNX accepts any channel window size for LRN and specifies that the window should extend more in the positive channel direction in case of even size: https://github.com/onnx/onnx/blob/master/docs/Operators.md#LRN

MXNet just throws an error when the size is even:
```
Check failed: param_.nsize % 2 == 1U (0 vs. 1) LRN only supports odd values for local_size
```
@sbodenstein

**Comment by eric-haibin-lin** [comment]
@haojin2 @anirudhacharya

**Comment by anirudhacharya** [comment]
@matteosal LRN indeed does accept only odd values for the window size, the LRN operator introduced in [this paper](https://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf) also prescribes the same. If the window size is even, then the local region over which normalization is performed will not be symmetric over the given element and will lead to erroneous results. 

If you look at the ONNX's test and example code, only odd number window sizes are being tested for - https://github.com/onnx/onnx/blob/master/docs/Operators.md#examples-31 It is a bug on the ONNX specification side that even valued window size is not being raised as an error. I will take it up with the ONNX community, and fix it.

**Comment by matteosal** [comment]
Why do you say that an even-sized window would lead to erroneous results? There is no fundamental reason why the window should be odd-sized. Setting an even-sized windows just gives a different, perfectly well defined operation.

MXNet doesn't comply with ONNX because it exposes a subset of the ONNX specification here. But of course limiting the ONNX prescription as you mention would be an alternative solution to extending the MXNet behaviour.

**Comment by kalyc** [comment]
@matteosal thanks for submitting this issue
@sandeep-krishnamurthy could you add label "Operator", "ONNX" to this?

**Comment by sandeep-krishnamurthy** [comment]
@spidyDev  @Roshrini - Any suggestions?

**Comment by spidyDev** [comment]
Couple of things to consider:
1. Does any of the other frameworks (TF, Pytorch, Caffe2, CNTK) supports even window size, if yes, then probably we should add  a "feature request" on MXNet
2. Are there multiple/enough use cases that uses  "even window size" to support this helps while training.

As for now, we should continue not supporting this if we do not have a valid use case for this operation.

**Comment by matteosal** [comment]
Pytorch has it, although with the opposite behavior to what ONNX prescribes, i.e. the window extends in the negative channel direction:
```
>>> import torch
>>> lrn2 = torch.nn.LocalResponseNorm(2, alpha=1, beta=1, k=0)
>>> lrn4 = torch.nn.LocalResponseNorm(4, alpha=1, beta=1, k=0)
>>> input = torch.ones(1, 6, 1, 1)
>>> lrn2(input)
tensor([[[[ 2.]],

         [[ 1.]],

         [[ 1.]],

         [[ 1.]],

         [[ 1.]],

         [[ 1.]]]])
>>> lrn4(input)
tensor([[[[ 2.0000]],

         [[ 1.3333]],

         [[ 1.0000]],

         [[ 1.0000]],

         [[ 1.0000]],

         [[ 1.3333]]]])

```

**Comment by matteosal** [comment]
For what is worth, this face recognition system (Lua Torch) uses it: https://github.com/cmusatyalab/openface

Project page: http://cmusatyalab.github.io/openface/

**Comment by anirudhacharya** [comment]
@matteosal thanks for bringing this to the community's notice. @sandeep-krishnamurthy please also tag this "Feature Request".

**Comment by vandanavk** [comment]
@mxnet-label-bot [Feature request]

**Comment by vandanavk** [comment]
@sandeep-krishnamurthy  Please change the label to [Feature request, Operator]
